refactor: make unintentional public spreadsheet api private or protected

ugur-vaadin opened the pull request:
## Description

There are public Flow Spreadsheet APIs that were not present in v8 Spreadsheet. 

This change converts the newly introduced public APIs into private or protected, depending on the usage within the package.

Fixes #3348

## Type of change

- [ ] Bugfix
- [ ] Feature
- [x] Refactor

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs-beta/latest/guide/contributing/overview/
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [ ] I have added tests to ensure my change is effective and works as intended.
- [x] New and existing tests are passing locally with my change.
- [x] I have performed self-review and corrected misspellings.

tomivirkki commented:
There's no need to keep any of the unused private API. I guess the methods were added for debugging purposes but not intended for the final product.

<img width="941" alt="Screenshot 2022-08-09 at 15 55 49" src="https://user-images.githubusercontent.com/1222264/183652381-374bc614-cd67-461b-89da-2abc3ea7a397.png">

Some of the methods referenced in the [issue](https://github.com/vaadin/flow-components/issues/3348) were not covered.

ugur-vaadin commented:
Removed unused private methods. 

Converted getColGroupingMax and getRowGroupingMax from public into protected.

The method getRowH is kept protected in order to maintain the consistency with getColW. It can still be made private though. What do you think?

Some of the methods mentioned in the issue are now used within the Spreadsheet class or other classes:
getCols
getColGroupingMax
getRowGroupingMax
isColGroupingInversed
isRowGroupingInversed
getDefRowH
getDefColW
getRowH
getColW
getSheetNames
getCellStyleToCSSStyle
getRowIndexToStyleIndex
getColumnIndexToStyleIndex
getLockedColumnIndexes
getLockedRowIndexes
getShiftedCellBorderStyles
getConditionalFormattingStyles
getVerticalScrollPositions
getHorizontalScrollPositions
getComponentIDtoCellKeysMap

tomivirkki commented:
> Some of the methods mentioned in the issue are now used in other classes in the package:

If removal isn't an option for some methods, rather make them package protected. No need to add this strange undocumented API available for users.

> Some of the methods referenced in the issue were not covered

The screenshot in the description only included a subset of the unintentional API. Please see the link in the description for full listing.

ugur-vaadin commented:
Addressed the previous comments.

Also deleted some unused fields.

One thing to note is while there is a public method `getDescription`, there is no setter for it. At this state, it can only return an empty string. Is this field still in use? If so, I think we should introduce a setter or any other way to set the description. If not, we can maybe deprecate it.

tomivirkki commented on vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow/src/main/java/com/vaadin/flow/component/spreadsheet/Spreadsheet.java:
```diff
@@ -283,538 +245,429 @@ public void setWidth(String width) {
     }
 
-    public float getDefRowH() {
+    protected float getDefRowH() {
```
> If removal isn't an option for some methods, rather make them package-private. No need to add this strange undocumented API available for users.

Why are these API still protected? Were they protected in V8 TabSheet?

ugur-vaadin replied:
The protected ones are now used in other classes. For this example, it is used in SpreadsheetFactory.

tomivirkki replied:
SpreadsheetFactory is in the same package as SpreadSheet, so it can be package-private, right? (to not expose this undocumented internal API for extensions)

ugur-vaadin replied:
Sure, done.

tomivirkki commented:
> One thing to note is while there is a public method getDescription, there is no setter for it. At this state, it can only return an empty string. Is this field still in use? If so, I think we should introduce a setter or any other way to set the description. If not, we can maybe deprecate it.

The description API was used in V8 to add a tooltip to components.

<img width="560" alt="Screenshot 2022-08-16 at 11 00 06" src="https://user-images.githubusercontent.com/1222264/184828997-cc376f02-7119-438a-b284-eebbbc8b30b4.png">

It's no longer there in the Flow components so it can be removed from SpreadSheet also.

ugur-vaadin commented:
The description field is now removed.
